lib/ssh.js

```diff
@@ -613,6 +617,7 @@ SSH2Stream.prototype.reset = function(noend) {
         kexinit: undefined,
         identRaw: undefined,
         rekeyQueue: [],
+        authsQueue: [],
```

Comment:
Since we're actually using `authsQueue` for both inbound for servers and outbound for clients, maybe we move it up a level so it's directly on the `_state` object. Also `authMethod` should be removed since the queue supersedes it.